tests/test_cache.py

```diff
@@ -0,0 +1,125 @@
+import os
+import pytest
+import tempfile
+
+from pipeline_migration.cache import (
+    ENV_FBC_DIR,
+    FBC_DIR_PREFIX,
+    FileBasedCache,
+    get_cache,
+    set_cache_dir,
+)
+from tests.utils import generate_digest
```

Comment:
For my understanding, how come the `tests.utils` module is importable even though we set `--import-mode=importlib` in pyproject.toml?

From the docs, it sounds like this shouldn't work https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html
> One drawback however is that test modules are non-importable by each other. Also, utility modules in the tests directories are not automatically importable because the tests directory is no longer added to [sys.path](https://docs.python.org/3/library/sys.html#sys.path).

Reply:
`tests` and even `src` can be treated as namespace packages.